sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/ForbidCertainMethodCheck.java

```diff
@@ -0,0 +1,267 @@
+        private final Integer lowerLimit;
+        /** Upper limit of the range. No upper limit when null. */
+        private final Integer upperLimit;
+
+        /**
+         * Initialize IntRange object with a lower limit and an upper limit.
+         * @param lowerLimit lower limit of the range, must be >= 0, null is equivalent to 0
+         * @param upperLimit upper limit of the range, null is equivalent to infinity
+         */
+        IntRange(Integer lowerLimit, Integer upperLimit) {
+            this.lowerLimit = lowerLimit;
+            this.upperLimit = upperLimit;
+        }
+
+        /**
+         * Create a range object corresponding to it string representation.
+         * @param range string representation of the range
+         * @return IntRange object for the string
+         *
+         * @throws CheckstyleException if the specified range is not valid
+         */
+        static IntRange from(String range) throws CheckstyleException {
+            Integer lowerLimit = null;
+            Integer upperLimit = null;
+            if (!range.contains("-")) {
+                lowerLimit = Integer.parseInt(range.trim());
+                upperLimit = lowerLimit;
+            }
+            else {
+                final Matcher matcher = RANGE_PATTERN.matcher(range);
+                if (!matcher.find()) {
+                    throw new CheckstyleException("Specified range is not valid: " + range);
+                }
+                final String lowerLimitString = matcher.group(1);
+                final String upperLimitString = matcher.group(2);
+                if (lowerLimitString.length() == 0 && upperLimitString.length() == 0) {
+                    throw new CheckstyleException("Specified range is unbounded on both side: "
+                        + range);
+                }
+                if (lowerLimitString.length() > 0) {
+                    lowerLimit = Integer.parseInt(lowerLimitString);
+                }
+                if (upperLimitString.length() > 0) {
+                    upperLimit = Integer.parseInt(upperLimitString);
+                }
+                if (lowerLimit != null && upperLimit != null && lowerLimit > upperLimit) {
+                    throw new CheckstyleException(
+                        "Lower limit of the range is larger than the upper limit: " + range);
+                }
+            }
+            return new IntRange(lowerLimit, upperLimit);
+        }
+
+        /**
+         * Check if range contain given number. Range is closed.
+         * If lower/upper bound is absent, it is considered unbounded on lower/upper side.
+         * @param num the number to be checked
+         * @return true if number is contained in the range, false otherwise
+         */
+        public boolean contains(int num) {
+            final boolean lowerLimitCheckPassed = lowerLimit == null || num >= lowerLimit;
+            final boolean upperLimitCheckPassed = upperLimit == null || num <= upperLimit;
```

Comment:
you should not have null in code, you can use 0 or MAX .

Reply:
Done. Also changed Integer to int.